lib/s3.js

```diff
@@ -141,4 +140,28 @@ module.exports = CoreObject.extend({
       });
     });
   },
+
+  listAllObjects: function(options) {
+    var client         = this._client;
+    var listObjects    = Promise.denodeify(client.listObjects.bind(client));
+    var allRevisions   = [];
+
+    function listObjectRecursively(options) {
+      return listObjects(options).then(function(response) {
+        [].push.apply(allRevisions, response.Contents);
+
+        if (response.IsTruncated) {
+          var nextMarker = response.NextMarker || response.Contents[response.Contents.length - 1].Key;
```

Comment:
Ok this seems to be only set when passing the `Delimiter`-option http://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/S3.html#listObjects-property. I think we can ignore `NextMarker` alltogether then.